### ISSUE TITLE
Thoughts on adding a scikit-learn compatible API?

### ISSUE THREAD

**Issue by jdwittenauer** [fchollet/keras#149]
Is there any interest in either modifying the existing model class or adding a "wrapper" class (I'm leaning towards the latter as the better option) to implement the required methods and attributes to make keras more compatible with scikit-learn?  To be clear, I'm referring to the requirements outlined [here](http://scikit-learn.org/stable/developers/#apis-of-scikit-learn-objects).  No additional dependencies would be necessary, and it would allow developers already familiar with scikit-learn to use keras out-of-the-box alongside existing code/models.  I don't think any of the current code would have to be touched to achieve this.

Thoughts?  I can spend some time looking at this if there's a decent chance it would be accepted.

**Comment by fchollet** [comment]
It wouldn't really be possible to make the model class sklearn-compatible, since Keras models work under very different constraints from sklearn. But it would definitely be possible to develop a sklearn-compatible wrapper that would give the user access to a subset of the functionality of the model class (under additional assumptions).

There are no immediate plans to build this, but the idea has been raised before... if you would like to give it a try, then you are welcome to do it, it would likely be accepted.

**Comment by alexjc** [comment]
This is a great idea! We just started prototyping `keras` integration into `scikit-neuralnetwork`:
https://github.com/aigamedev/scikit-neuralnetwork

I'm wondering if I could make an argument for keeping the wrapper separate/compatible, for a couple reasons:
* All NN implementations in Python could be wrapped under compatible APIs, helping the best implementation "win".  Our sklearn-compatible API has been through a couple iterations already.
* It's useful for the Python community to help compete with Lua's advantage, also providing some stability.  Most users just want to get a package from PIP and have it just work, as we found out.
* I think end-user support requests may be better off separate from the main project; most of the issues we had so far never needed to propagate upward to the underlying library...

I don't mean to hijack the thread, just adding my thoughts. I hope this happens either way :-)

**Comment by mbatchkarov** [comment]
Is this functionality still supported? I can't see any of the tests in the current master.

Also, it would be nice if models supported more of the sklearn workflow. For example, `KerasClassifier` cannot be used in `cross_val_score` because the underlying model cannot be cloned (because its `get_params` method does something different to what `sklearn` expects). Are there any plans to address this?

**Comment by gustavodemari** [comment]
+1 on @mbatchkarov comment

**Comment by fchollet** [comment]
There is a PR for this functionality here:
https://github.com/fchollet/keras/pull/1225/files

It wasn't merged because it looks to me like a big hack. If you think of a
cleaner, simpler way to do it, please submit a new PR.
